### ISSUE TITLE
Add test for SimpleReflexAgent

### ISSUE THREAD

**Issue by apb7** [aimacode/aima-python#768]
Hello, @norvig and @MrDupin! I would like to add tests for the `SimpleReflexAgentProgram` in `test_agents.py`.  
Also, I had a doubt regarding this. Currently, we have a `rule_match` function which looks like this:  
```python
def rule_match(state, rules):
    """Find the first rule that matches state."""
    for rule in rules:
        if rule.matches(state):
            return rule
```  
We do not have a `matches` function for the `rule`. We need to design this function before we go for the test. How should we go about this? We can create a `Rule` class in `test_agents.py` only without changing `agents.py` for the testing part. Please suggest!  
Thanks!

**Comment by apb7** [comment]
Hello, @norvig! Can you please take a look at this?  
Thanks!

**Comment by apb7** [comment]
@MrDupin: Can you please suggest a possible way for this? 
Thanks!

**Comment by MrDupin** [comment]
In #751, Dr. Norvig said that he doesn't like an implementation of `Rule`. I assume this means the class should be implemented on each individual case, so you need to create the class in your tests. So, `rules` is a list of objects of type `Rule`, where `Rule` is a class that needs to be implemented for this particular case.

I think you should add the class to `test_agents` and also maybe add some documentation of the above (in a nutshell, explain that `rules` is a list of `Rule` objects). That would be really useful!

**Comment by apb7** [comment]
Okay, got it! I think it would be better if we create a `Rule` class inside the test_SimpleReflexAgentProgram function in test_agents. I'll add the documentation for the class as well. How does this sound?  
Thanks!

**Comment by MrDupin** [comment]
Yes, sounds good.

For the documentation, it should go in the `agents.py` script. That way future contributors will know that they need to create a `Rule` class on their own. You don't have to write much, just mention it in a couple of lines.

**Comment by apb7** [comment]
Okay, understood. Thanks, @MrDupin!

**Comment by apb7** [comment]
@MrDupin: This issue has been addressed in PR #808> Therefore closing it.
